frp_deploy.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+#Set flags for hot update.
+
+frps_FLAG=`ps -ef |grep -w frps|grep -v grep|wc -l`
+frpc_FLAG=`ps -ef |grep -w frpc|grep -v grep|wc -l`
+
+#Force stop services before upgrade.
+
+systemctl stop frps
+
+systemctl stop frpc
+
+#Detect the arch.
+
+case $(uname -m) in
+    "x86_64")
+        ARCH="amd64"
+        ;;
+    "i*86")
+        ARCH="386"
+        ;;
+    "arm64")
+        ARCH="arm64"
+        ;;
+    "armhf")
+        ARCH="arm"
+        ;;
+    "mips64")
+        ARCH="mips64"
+        ;;
+    "mips64le")
+        ARCH="mips64le"
+        ;;
+    "mips")
+        ARCH="mips"
+        ;;
+    *)
+        echo "Unsupport architecture." && exit 0
+esac
+echo "Detected ${ARCH} architecture."
+#Get latest release tag
+RELEASETAG=$(git ls-remote --tags https://github.com/fatedier/frp | sort -t '/' -k 2 -V | sed -e '/\^/d')
+export FRP_VERSION=${RELEASETAG:0-6}
+echo "The latest frp release is ${FRP_VERSION}"
+mkdir -p /etc/frp
+
+rm -rf /tmp/frp_${FRP_VERSION}_linux_${ARCH}*
+cd /tmp
+ echo "Downloading tar package from Github......"
+ wget -q "https://github.com/fatedier/frp/releases/download/v${FRP_VERSION}/frp_${FRP_VERSION}_linux_${ARCH}.tar.gz"
+ echo "Download complete."
+ echo "Unpacking......"
+ tar xzf frp_${FRP_VERSION}_linux_${ARCH}.tar.gz
+ echo "Installing binary files......"
+ cp ./frp_${FRP_VERSION}_linux_${ARCH}/frps /usr/bin && chmod +x /usr/bin/frps
+ cp ./frp_${FRP_VERSION}_linux_${ARCH}/frpc /usr/bin && chmod +x /usr/bin/frpc
+ setcap cap_net_bind_service=ep /usr/bin/frps #Give frps binary access to well-known ports(smbd,NetBIOS,etc).
+ echo "Installing configuration files......"
+ test -e /etc/frp/frpc.ini && echo "Configuration files exist. Do not copy templetes." || (echo "Configuration files do not exist. Copy templetes." && cp ./frp_${FRP_VERSION}_linux_${ARCH}/frp*.ini /etc/frp)
+ echo "Configuration file is installed to /etc/frp ."
+
+ #Fix systemctl warning.
+ cp ./frp_${FRP_VERSION}_linux_${ARCH}/systemd/* /lib/systemd/system
```

Comment:
Not all linux os support systemd?

Reply:
And we can copy systemd configurefiles to this repo, it will not managed in frp repo.

Reply:
> Not all linux os support systemd?
Gentoo / Slackware are affected. For gentoo, we can use systemd over the openRC. For slackware, I don't have such environment / knowledge to test.

Reply:
> And we can copy systemd configurefiles to this repo, it will not managed in frp repo.

Not necessarily,Keeping systemd unit files with binaries can make install script simpler, or we need to download them from this repo separately.

Reply:
> > And we can copy systemd configurefiles to this repo, it will not managed in frp repo.
> 
> Not necessarily,Keeping systemd unit files with binaries can make install script simpler, or we need to download them from this repo separately.

Upload the init.d script to frp repo?

Reply:
The process should be:
1. Detect system arch and os.
2. Download corresponding scripts(systemd configs)  or others.
3. Download binary files.
4. Move files to correct locations.
5. Display usage message.

We can only support specified os you tested and comment it.